### PR TITLE
#28 feature: clear-cart

### PR DESCRIPTION
**Pull request by vanessa-abewe**
#### What does this PR do?

This PR introduces the action to clear the cart within the application.  allowing users to delete products from their cart.

#### Description of Task to be completed?
The task involves implementing the necessary components and logic to handle clearing cart. This includes deleteting cart without leaving the current page

#### How should this be manually tested?
```bash
git clone https://github.com/atlp-rwanda/ecommerce-app-predators-fe.git
```
```bash
cd ecommerce-app-predators-fe
```
```bash
git checkout branch-name
```
```bash
npm i
```
```bash
npm run dev
```

#### What are the relevant pivotal tracker/Trello stories?
Trello [Link](https://trello.com/c/D6X0U9aC) (#-28)

#### Screenshots (if appropriate)

### CODE REVIEW

**Review by ALVINdimpos — APPROVED** [review]

File: src/App.tsx [hunk 1]
```diff
@@ -25,12 +25,15 @@ import 'react-toastify/ReactToastify.css';
 
 import NotFound from './pages/error/NotFound';
 import WishlistPage from './pages/WishlistPage';
+//import FetchCarts from '../src/pages/cart/FetchCart';
```
Comment (chanceDira) [inline src/App.tsx]: remove these lines

File: src/App.tsx [hunk 2]
```diff
@@ -62,6 +62,14 @@ function App() {
         <Route path="*" element={<NotFound/>} />
 
         <Route path="*" element={<NotFound/>} />
+        {/* <Route path="/cart" element={<FetchCarts />} /> */}
```
Comment (chanceDira) [inline src/App.tsx]: remove this line

File: src/components/Buyer/HeaderNav.tsx [hunk 3]
```diff
@@ -83,4 +83,4 @@ function NavHeader({ onSearchText }: NavHeaderProps) {
 
 export default NavHeader;
 
-// console.log('fixing')
\ No newline at end of file
+// console.log('fixing')
```
Comment (chanceDira) [inline src/components/Buyer/HeaderNav.tsx]: remove this comment too

File: src/pages/cart/cartData.tsx [hunk 4]
```diff
@@ -8,7 +10,7 @@ import {
   deleteCart,
   deleteAllCarts,
   updateCartQuantity,
-} from '../../redux/action/cartAction';
+} from '../../redux/action/cartsAction';
 
 interface CartItem {
   [x: string]: any;
```
Comment (chanceDira) [inline src/pages/cart/cartData.tsx]: Can you explain why, you are using "any" here?

File: src/pages/viewProduct.tsx [hunk 5]
```diff
@@ -11,7 +11,7 @@ import StaticReview from '../components/StaticReview';
 import { FaCheck } from 'react-icons/fa';
 function ViewProduct() {
   const { id } = useParams(); // Get the product ID from the URL params
-  const selectedProduct = useSelector(
+   const selectedProduct = useSelector(
     (state: any) => state.CollectionProducts.selectedProduct
```
Comment (chanceDira) [inline src/pages/viewProduct.tsx]: use interface here too

File: src/redux/reducers/productReducer.tsx [hunk 6]
```diff
@@ -1,6 +1,6 @@
 import { createSlice} from "@reduxjs/toolkit";
 import {  fetchProductsCollection } from "../action/productActions";
-
+import {fetchProductById} from "../action/ProductAction"
 export interface Product {
   product: any;
```
Comment (chanceDira) [inline src/redux/reducers/productReducer.tsx]: no "any"s

**Review by chanceDira — COMMENTED** [review]
Looking good, just some code cleaning.

**Review by faid-terence — APPROVED** [review]

**Review by NTElissa — APPROVED** [review]

**Pull request merged**